wget/build.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+set -x
+
+WGET_VERSION=1.19.2
+OPENSSL_VERSION=1.1.0g
+
+function build_openssl() {
+    cd /build
+
+    # Download
+    curl -LO https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz
+    tar zxvf openssl-${OPENSSL_VERSION}.tar.gz
+    cd openssl-${OPENSSL_VERSION}
+
+    # Configure
+    CC='/usr/bin/gcc -static' ./Configure no-shared no-async linux-x86_64
+
+    # Build
+    make -j4
+    echo "** Finished building OpenSSL"
+}
+
+function build_wget() {
+  cd /build
+
+  # Download
+  curl -LO https://ftp.gnu.org/gnu/wget/wget-${WGET_VERSION}.tar.gz
+  tar -zxvf wget-${WGET_VERSION}.tar.gz
+  cd wget-${WGET_VERSION}
+
+  # Build
+  CC='/usr/bin/gcc -static' \
+    CFLAGS='-fPIC' \
+    CPPFLAGS="-I/build" \
+    LDFLAGS="-L/build/openssl-${OPENSSL_VERSION}" \
+    OPENSSL_CFLAGS="-I/build/openssl-${OPENSSL_VERSION}/include" \
+    OPENSSL_LIBS="-lssl -lcrypto" \
+    ./configure \
+    --with-ssl=openssl
```

Comment:
Actually it looks like it builds statically but links dynamically.  It seems to run on my target system because I'm replacing a broken wget and happen to have the right deps.  I'm not quite sure how to force the linker to do the right thing.

Reply:
ldd on the resulted bin